src/components/buttons/ConnectWallet/index.tsx

```diff
@@ -1,93 +1,80 @@
-        const ready = mounted && authenticationStatus !== 'loading';
-        const connected =
-          ready && account && chain && (!authenticationStatus || authenticationStatus === 'authenticated');
+  // walletChainId is the chainId available on the wallet level
+  const { address, chainId: walletChainId } = useAccount();
+  const { isConnected } = useAppKitAccount();
+  // appkitNetwork contains the chainId currently configured on the app level
+  const { caipNetwork: appkitNetwork, switchNetwork } = useAppKitNetwork();
+  const { open } = useAppKit();
 
-        return (
-          <div
-            {...(!ready && {
-              'aria-hidden': true,
-              style: {
-                opacity: 0,
-                pointerEvents: 'none',
-                userSelect: 'none',
-              },
-            })}
-          >
-            {(() => {
-              if (!connected) {
-                return (
-                  <button
-                    onClick={() => {
-                      openConnectModal();
-                      handleUserClickWallet();
-                    }}
-                    type="button"
-                    className="btn-vortex-secondary btn rounded-3xl group"
-                  >
-                    <p className="flex">
-                      Connect <span className="hidden lg:block lg:ml-1">Wallet</span>
-                    </p>
-                    <PlayCircleIcon className="w-5 group-hover:text-pink-600" />
-                  </button>
-                );
-              }
+  // Check if the network selected in the wallet extension is enabled in our wagmi config
+  const isOnSupportedNetwork = wagmiConfig.chains.find((chain) => chain.id === walletChainId) !== undefined;
 
-              if (chain.unsupported) {
-                return (
-                  <button
-                    onClick={() => {
-                      openChainModal();
-                      handleUserClickWallet();
-                    }}
-                    type="button"
-                    className="btn-vortex-secondary btn rounded-3xl group"
-                  >
-                    Wrong network
-                    <PlayCircleIcon className="w-5 group-hover:text-pink-600" />
-                  </button>
-                );
-              }
+  const ConnectButton = useMemo(() => {
+    if (!isConnected) {
+      return (
+        <button
+          onClick={() => {
+            handleUserClickWallet();
+            open({ view: 'Connect' });
+          }}
+          type="button"
+          className="btn-vortex-secondary btn rounded-3xl group"
+        >
+          <p className="flex">
+            Connect <span className="hidden lg:block lg:ml-1">Wallet</span>
+          </p>
+          <PlayCircleIcon className="w-5 group-hover:text-pink-600" />
+        </button>
+      );
+    } else if (!isOnSupportedNetwork) {
+      return (
```

Comment:
We could extract these buttons and then just simply do:
`else if (!isOnSupportedNetwork) return <ButtonWrongNetwork/>`

Reply:
We could but do you think it's worth it? All three buttons differ in more than just text as they also have different stylings.

Reply:
It improves the readability but it is optional, that's why I approved 😄